### DISCUSSION TITLE
Inquiries about using joysticks while driving on their own.

### DISCUSSION THREAD

**Discussion by jakhome1** [prl-mushr/mushr#63] [Software]
Dear Sir
As far as I know, you have to press Deadman 5(R1) to drive autonomously.
I wonder where you have to press deadman 5 in the source to drive autonomously.
Thank you and best regards,

**Comment by schmittlema** [comment]
The deadman button location is defined in the following configuration file:
https://github.com/prl-mushr/mushr_base/blob/master/mushr_base/config/joy_teleop.yaml#L46

The rest of the file has other configurations for teleop, etc. if you would like to tweak those as well

**Reply by jakhome1** [reply]
Can't I know the part where autonomous driving starts only when I press the R1 button in the navigation source file?

**Reply by schmittlema** [reply]
How it works is the mux chooses which commands are sent to the robot by a preset priority [set here](https://github.com/prl-mushr/mushr_base/blob/master/ackermann_cmd_mux/param/mux.yaml). Autonomy commands are constantly being sent (if autonomy running) but they are overruled (higher priority) by teleop commands. When you press R1 you are blocking teleop commands from being sent which then allows the mux to forward the autonomy commands. Specifically [here](https://github.com/prl-mushr/mushr_base/blob/master/mushr_base/src/joy_teleop.py#L82) only one command from joy can be processed at one time so when you press R1 it blocks other commands from being processed (as described in the first file I sent). I hope this helps, if you have a specific thing you are trying to do I can provide a bit more advice.

**Reply by jakhome1** [reply]
I want to run autonomous driving with a button other than the R1 button.

**Reply by schmittlema** [reply]
Oh in that case you would just change the number to another button in the joy_teleop.yaml file. I believe the button numbers are associated with an index in the button array in the /car/joy topic. So run the stack, `rostopic echo /car/joy` and press the button you want to use. The index that changes is the button you want to use, I think.
